### ISSUE TITLE
Lax commadangle

### ISSUE THREAD

**Issue by bfred-it** [sindresorhus/eslint-config-xo#29]
ESLint 3.0.0 just [disabled this rule](http://eslint.org/docs/user-guide/migrating-to-3.0.0#changes-to-eslintrecommended) in their recommended config:

> [comma-dangle](http://eslint.org/docs/rules/comma-dangle) used to be recommended because Internet Explorer 8 and earlier threw a syntax error when it found a dangling comma on object literal properties. However, [Internet Explorer 8 was end-of-lifed](https://www.microsoft.com/en-us/WindowsForBusiness/End-of-IE-support) in January 2016 and all other active browsers allow dangling commas. As such, we consider dangling commas to now be a stylistic issue instead of a possible error.

I'd suggest to follow suit and drop the error. Dangling commas are supported in arrays, object, and hopefully soon even in [function parameters](http://www.2ality.com/2015/11/trailing-comma-parameters.html). They also make git diffs smaller:

```js
[
	"one"
]
```

add "two":

```diff
[
-	"one"
+	"one",
+	"two",
]
```

but with a dangling comma:

```js
[
	"one",
]
```

becomes:

```diff
[
	"one",
+	"two",
]
```

For this reason I'd prefer the [option](http://eslint.org/docs/rules/comma-dangle#options) `["error", "always-multiline"]` but `["error", "only-multiline"]` would be _lax_ enough.

**Comment by jamestalmage** [comment]
Personally, I'm 👎 .

I find comma dangling to be weird, and I don't feel saving the occasional line saved in a diff is worth peppering my code with weirdness.  ESLint only removed it from the recommended config because it's changed from a potential error to an entirely stylistic choice. XO enforces tons of stylistic choices on it's users.

If we **do** decide to do this, then we should go all in with `always-multiline`. Allowing a mixture in your codebase is just anarchy.

You can always override this in your `package.json`:

```js
{
  "xo": {
    "rules" : {
      "comma-dangle": ["error", "always-multiline"]
    }
  }
```

**Comment by bfred-it** [comment]
Yeah, I'm fine with stylistic choices, I'm only hoping that it was done to avoid browser errors

**Comment by SamVerschueren** [comment]
I'm not a fan of dangling comma's either so I'm 👎 on this one. But I'm always open for discussion :).

**Comment by bfred-it** [comment]
Dangling commas might look a bit weird, it's true, but I thought of this: (forget code style preconceptions for a moment)

```js
// pardon the lack of spaces
[0,0,0,0]
```

^ No dangling comma. Looks beautiful, commas are symmetrical and natural.

```js
[0,0,0,0,]
```

^ hmmm. So-so.

```js
[
	0
]
```

^ Multiline, single element. You wouldn't do that it's still beautiful, symmetrical and it's currently valid in xo.

```js
[
	0,
	0,
	0,
	0,
	0
]
```

^ Multiline, multiple elements, no dangle. That final element feels off. It doesn't match the lines above. It looks like you forgot the period at the end of a sentence or the semicolon at the end of a line.


```js
[
	0,
	0,
	0,
	0,
	0,
]
```

^ Read them line by line and forget the brackets. Finally they all match! 🎉 

So:

```js
// ❌
[0,0,0,0,]

// ✅
[
	0,
	0,
	0,
	0,
	0,
]
```

Now, the only issue is that `always-multiline` means that this also needs a comma:

```js
[
	0,
]
```

That's why I'd go for `only-multiline` or perhaps suggest an option like `always-multiline-multielements` to eslint

**Comment by lil5** [comment]
@SamVerschueren could you explain why?

**Comment by lil5** [comment]
@bfred-it 
always-mulitline is there specifically against diffs
The only reason someone would have a single value multi line array or object is for later additions.
Exactly the reason for it.

```js
let array = [
  'world!',
];
```

Later...
```js
let array = [
  'Hello',
  'world!',
];
```

The only change will be of adding `'Hello',` as it should be.
Styling should not constantly be haunting git commits.

**Comment by bfred-it** [comment]
True. If the element is one and expected to stay one, perhaps it should just be written as

```js
let array = ['world!'];
```

**Comment by lil5** [comment]
@bfred-it you could also make it a warning; `"warn"`

**Comment by kevva** [comment]
I'm 👎 on `"comma-dangle": ["error", "always-multiline"]` or disabling it since I'd still want `xo` to tell me if I've left a comma dangle somewhere.

**Comment by SamVerschueren** [comment]
I'm with @kevva on this one.

**Comment by bfred-it** [comment]
@kevva the point of this issue is to change your opinion on whether you should use the dangle or not.

Besides git, this is also what you can do: 

- move items freely
- remove items freely

![ewe](https://user-images.githubusercontent.com/1402241/30582754-0cef1b9a-9d4f-11e7-89f0-519cc15c62af.gif)

- Consistently reach the end of the line without including the comma

![232323](https://user-images.githubusercontent.com/1402241/30582858-5cbd1564-9d4f-11e7-8154-3db0c894f167.gif)

There are many reason to have dangling commas, yet we're stuck with "I don't like the way it looks"
